pkgs/applications/video/video-trimmer/default.nix

```diff
@@ -0,0 +1,93 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, rustPlatform
+, gnome
+, pkg-config
+, meson
+, wrapGAppsHook4
+, appstream-glib
+, desktop-file-utils
+, blueprint-compiler
+, ninja
+, python3
+, gtk3-x11
+, glib
+, gobject-introspection
+, gtk4
+, libadwaita
+, gst_all_1
+}:
+
+stdenv.mkDerivation rec {
+  pname = "video-trimmer";
+  version = "0.7.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "YaLTeR";
+    repo = "video-trimmer";
+    rev = "v${version}";
+    sha256 = "sha256-D7wjJkdqqjjwwYEUZnNr7hFQK59wfTnaCLXCy+SK8Jo=";
+  };
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-cB5dVrEbISvHrOb87uVZSkT694VKtPtyk+c1tYNCTp0=";
+  };
+
+  patches = [
+    # The metainfo.xml file has a URL to a screenshot of the application,
+    # unaccessible in the build's sandbox. We don't need the screenshot, so
+    # it's best to remove it.
+    ./remove-screenshot-metainfo.diff
+  ];
+
+  postPatch = ''
+    patchShebangs \
+      build-aux/meson/postinstall.py \
+      build-aux/cargo.sh \
+      build-aux/dist-vendor.sh
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+    meson
+    wrapGAppsHook4
+    appstream-glib
+    desktop-file-utils
+    blueprint-compiler
+    ninja
+    # For post-install.py
+    python3
+    gtk3-x11 # For gtk-update-icon-cache
```

Comment:
This should be just `gtk3`. But ideally, they would switch to meson’s `gnome.post_install` which supports GTK 4 as well, see https://github.com/NixOS/nixpkgs/pull/179817#discussion_r913187375.